### PR TITLE
[JBIDE-22763] use explorer selection if no image is selected

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

**Comment by adietish** [comment]
This was about adding logic back in that went lost in the last refactoring. This now should match the prior state. 
@fbricon @jeffmaury @bdshadow @scabanovich please review

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeployImageHandler.java [hunk 1]
```diff
@@ -49,10 +49,14 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 		IProject project = null;
 		ISelection selection = HandlerUtil.getCurrentSelection(event);
 		final IDockerImage image = UIUtils.getFirstElement(selection, IDockerImage.class);
-		if (image != null) {
-			selection = OpenShiftUIUtils.getOpenShiftExplorerSelection();
+		if (image == null
+				|| OpenShiftUIUtils.hasOpenShiftExplorerSelection()) {
```
Comment (scabanovich) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeployImageHandler.java]: May it be && rather than || ?

Comment (adietish) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeployImageHandler.java]: +1

**Comment by fbricon** [comment]
+1

**Pull request merged**